resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala

```diff
@@ -787,6 +792,15 @@ private[spark] class MesosClusterScheduler(
     }
   }
 
+  private def taskIsOutdated(taskId: String): Boolean = {
+    if (retriedDrivers.contains(taskId)) {
+      true
+    } else {
+      retriedDrivers.add(taskId)
```

Comment:
having state mutation logic in a boolean function which checks some condition looks a bit confusing